### PR TITLE
Replaced clone calls in linalg (fixes #5014)

### LINKED ISSUE #5014: remove `.clone` in linalg methods
Linalg methods that return newly allocated vectors sometimes clone an input vector, instead of just allocating empty memory. That is unnecessary and could be changed.

### PR DESCRIPTION
**Pull request by gf712**

### CODE REVIEW

File: src/shogun/lib/SGMatrix.cpp [hunk 1]
```diff
@@ -407,6 +407,12 @@ T* SGMatrix<T>::clone_matrix(const T* matrix, int32_t nrows, int32_t ncols)
 	return result;
 }
 
+template <class T>
+SGMatrix<T> SGMatrix<T>::empty_like(const SGMatrix<T>& other)
```
Comment (karlnapf) [inline src/shogun/lib/SGMatrix.cpp:411]: what about just `empty`?

**Review by karlnapf — APPROVED** [review]

File: src/shogun/lib/SGMatrix.cpp (hunk 1 above)
Comment (gf712) [reply src/shogun/lib/SGMatrix.cpp:411]: sure, this was just to keep it in line with numpy

Comment (karlnapf) [reply src/shogun/lib/SGMatrix.cpp:411]: ah I see. Think that is good

Comment (gf712) [reply src/shogun/lib/SGMatrix.cpp:411]: shall I merge this like it is now?

**Pull request merged**